src/PHPStan/PregMatchTypeSpecifyingExtension.php

```diff
@@ -67,7 +71,12 @@ public function specifyTypes(MethodReflection $methodReflection, StaticCall $nod
             return new SpecifiedTypes();
         }
 
-        $matchedType = $this->regexShapeMatcher->matchExpr($patternArg->value, $flagsType, TrinaryLogic::createFromBoolean($context->true()), $scope);
+        if (stripos($methodReflection->getName(), 'matchAll') !== false) {
+            $matchedType = $this->regexShapeMatcher->matchAllExpr($patternArg->value, $flagsType, TrinaryLogic::createFromBoolean($context->true()), $scope);
```

Comment:
Same here.. Probably easiest to bump the conflict rule tbh, not worth having ugly code here.

Reply:
Ok I did this myself in a follow-up commit. Thanks for the PR! I guess preg_replace_callback could also be done now with phpstan 1.11.10?

Reply:
Yep https://github.com/phpstan/phpstan-src/pull/3281